Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/CommonComponents/LocationDrawingsButton.tsx

```diff
@@ -0,0 +1,163 @@
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  01/06/2024 - Collins Self
+//       Generated original version of source code.
+//
+//******************************************************************************************************
+import React from 'react';
+import { BtnDropdown, GenericController, LoadingScreen, Modal, ServerErrorIcon, ToolTip } from '@gpa-gemstone/react-interactive';
+import { OpenXDA } from '@gpa-gemstone/application-typings';
+import { CrossMark } from '@gpa-gemstone/gpa-symbols';
+import LocationDrawingsTable from '../Location/LocationDrawingsTable';
+
+interface LocationDrawingsButtonProps {
+    Locations: OpenXDA.Types.Location[];
+    IsLoadingLocations?: boolean;
+}
+
+type DropDownOption = {
+    Label: JSX.Element | string,
+    Callback: () => void,
+    Disabled: boolean
+    ToolTipContent: JSX.Element,
+    ShowToolTip: boolean,
+    ToolTipLocation: ('top' | 'bottom' | 'left' | 'right'),
+    Key: string | number
+}
+
+const isValid = (location: OpenXDA.Types.Location, drawingData) => {
+    let e = "";
+
+    if (location == null
+        || (location.Alias === ""
+            && location.Description === ""
+            && location.ID === 0
+            && location.Latitude == null
+            && location.LocationKey === ""
+            && location.Longitude == null
+            && location.Name === ""))
+        e = 'No location(s) have been set.';
+    else if (drawingData.TotalRecords == 0)
+        e = 'No drawing(s) associated with location.';
+    return e;
+}
+
+const LocationDrawingsButton: React.FC<LocationDrawingsButtonProps> = (props) => {
+    const [hover, setHover] = React.useState<'none' | 'drawings'>('none');
+    const [pageState, setPageState] = React.useState<"loading" | "error" | "idle">("idle");
+    const [selectedLocation, setSelectedLocation] = React.useState<OpenXDA.Types.Location>();
+    const multipleLocations = React.useMemo(() => props.Locations.length > 1, [props.Locations]);
+    const [showDrawingsModal, setShowDrawingsModal] = React.useState<boolean>(false);
+    const [locationOptions, setLocationOptions] = React.useState<DropDownOption[]>([]);
+    const LocationDrawingController = new GenericController(`${homePath}api/LocationDrawing`, "Name", true);
+
+    React.useEffect(() => { // Generates the map of errors for each location
+        setPageState('loading');
+        setLocationOptions([]);
+
+        const handles = props.Locations.map((location, i) => {
+            if (location == null)
+                return null;
+            const handle = LocationDrawingController.PagedSearch([], 'Name', true, 1, location.ID)
```

Comment:
Why are we paging this? There appears to be no way to manipulate what page we are on.

Reply:
Also, it appears you don't understand how parentID really works. ParentID is if there is a parent table to this table.
Why are we even doing this lookup if we already have the location information from the prop?

Reply:
updated in latest commit